### PR TITLE
setup semantic-release to manage releases to npm

### PR DESCRIPTION
**Pull request by gurpreetatwal**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,41 +4,34 @@ language:
...
+      node_js: '10'
+      script:
+        - npm test
+        - npm run cover
+          # need a local clone to let semantic-release git push --dry-run
+        - git clone --bare https://github.com/smartcar/javascript-sdk.git
+        - npx semantic-release --branch "$TRAVIS_BRANCH" --extends ./build/sr-configs/verify.js
+    - stage: publish
+      node_js: '10'
+      if: branch = master
```
Comment (sankethkatta) [inline .travis.yml]: is this different than:
```yaml
on:
    branch: master
```

File: package.json [hunk 2]
```diff
@@ -1,38 +1,34 @@
...
   "scripts": {
-    "test": "npm run build && jest",
+    "pretest": "gulp build",
+    "test": "jest",
     "posttest": "npm run lint -s",
     "lint": "eslint . --cache",
     "cover": "npm test -- --coverage",
-    "build": "gulp build",
-    "publish": "gulp publish",
+    "release": "semantic-release --branch \"$(git rev-parse --abbrev-ref HEAD)\" --extends ./build/sr-configs/local.js",
```
Comment (sankethkatta) [inline package.json]: This seems confusing being named `"release"`. I think something like `"plan-relase"` or `"prepare-release"` or something else to the effect that its not actually releasing would be better.

File: build/verify-release.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  # Execute command, if it fails, exec fail
+  "${COMMAND[@]}" > /dev/null || fail "$FILE"
+}
+
+# Print
+echo "Checking if all files have version: $NEXT_RELEASE"
+
+# Ensure that package.json is in sync
+try "package.json" test "$(jq --raw-output .version package.json)" = "$NEXT_RELEASE"
+try "package-lock.json" test "$(jq --raw-output .version package.json)" = "$NEXT_RELEASE"
```
Comment (sankethkatta) [inline build/verify-release.sh]: is `jq` on travis already?

File: doc/README.md [hunk 4]
```diff
@@ -45,7 +45,7 @@ Initializes Smartcar class.
 | options.redirectUri | <code>String</code> |  | the registered redirect uri of the application |
 | [options.scope] | <code>Array.&lt;String&gt;</code> |  | requested permission scopes |
 | [options.onComplete] | [<code>OnComplete</code>](#OnComplete) |  | called on completion of auth flow |
-| [options.development] | <code>Boolean</code> | <code>false</code> | launch Smartcar auth in development mode to enable the mock vehicle brand |
+| [options.testMode] | <code>Boolean</code> | <code>false</code> | launch the Smartcar auth flow in test mode |
```
Comment (sankethkatta) [inline doc/README.md]: was this not done in the `testMode` PR?

File: build/release.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
```
Comment (sankethkatta) [inline build/release.sh]: point out that this only runs locally

File: build/release.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+NEXT_RELEASE="$1"
+
+node -e "
+  const file = './package.json'
+  const pkg = require(file);
+  pkg.version = '$NEXT_RELEASE';
+  fs.writeFileSync(file, JSON.stringify(pkg, null, 2) + '\\n')
```
Comment (sankethkatta) [inline build/release.sh]: semicolon

File: build/release.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+"
+node -e "
+  const file = './package-lock.json'
+  const pkg = require(file);
+  pkg.version = '$NEXT_RELEASE';
+  fs.writeFileSync(file, JSON.stringify(pkg, null, 2) + '\\n')
```
Comment (sankethkatta) [inline build/release.sh]: semicolon

File: build/release.sh [hunk 8]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (sankethkatta) [inline build/release.sh]: rename to roughly "update package version.js"

File: build/sr-configs/local.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+ * readme
+ */
+module.exports = {
+  ci: false,
+  dryRun: true,
+  repositoryUrl: 'git@github.com:smartcar/javascript-sdk',
+  verifyConditions: [],
+  verifyRelease: {
+    path: '@semantic-release/exec',
+    cmd: './build/release.sh ${nextRelease.version}',
```
Comment (sankethkatta) [inline build/sr-configs/local.js]: add in `npm run readme && npm run docs`

**Comment by sankethkatta** [comment]
Also can you add a `README.md` to the `/build` folder

File: build/verify-release.sh [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  printf "\033[1;31m[error]\033[0m please run \"npm run release\" %s does not have correct version\n" "$WHICH"
+}
+
+# Runs a command and if the command fails, prints that file is bad.
+#
+# @param FILE - name of file being tested
+# @param ...args - command to run
+try() {
+  local 'FILE'
+  local 'COMMAND'
```
Comment (RogWilco) [inline build/verify-release.sh]: See above comment about local variables.

File: .travis.yml (hunk 1 above)
Comment (gurpreetatwal) [reply .travis.yml]: no sir, this is the new conditonal syntax https://docs.travis-ci.com/user/conditional-builds-stages-jobs/

File: build/verify-release.sh (hunk 3 above)
Comment (gurpreetatwal) [reply build/verify-release.sh]: si, https://docs.travis-ci.com/user/build-environment-updates/2014-12-09/#2014-12-09

Comment (gurpreetatwal) [reply build/verify-release.sh]: si, https://travis-ci.com/smartcar/javascript-sdk/builds/88473284#L55

File: doc/README.md (hunk 4 above)
Comment (gurpreetatwal) [reply doc/README.md]: nope, I forgot to run docs

**Comment by RogWilco** [comment]
LGTM to me :+1:

File: build/verify-release.sh [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+LAST_RELEASE="$2"
+
+EXIT_CODE=0
+
+# Prints that $FILE is out of sync
+#
+# @param FILE - name of file out of sync
+fail() {
+  local 'WHICH'
+  EXIT_CODE=1
```
[the remark this answers is not in the document]
Comment (gurpreetatwal) [reply build/verify-release.sh]: I was not able to make this work correctly so I reverted the change

**Pull request merged**